Framework/Algorithms/test/CalculateEfficiencyTest.h

```diff
@@ -233,14 +227,86 @@ class CalculateEfficiencyTest : public CxxTest::TestSuite {
     TS_ASSERT(ws2d_out->getDetector(1826)->isMasked())
     TS_ASSERT(ws2d_out->getDetector(2014)->isMasked())
     TS_ASSERT(ws2d_out->getDetector(2015)->isMasked())
+  }
+  Mantid::API::AnalysisDataService::Instance().remove(inputWS);
+  Mantid::API::AnalysisDataService::Instance().remove(outputWS);
+}
+}
+
+class CalculateEfficiencyTest : public CxxTest::TestSuite {
+public:
+  void tearDown() {}
+  void testName() { TS_ASSERT_EQUALS(correction.name(), "CalculateEfficiency") }
+
+  void testVersion() { TS_ASSERT_EQUALS(correction.version(), 1) }
+
+  void testInit() {
+    TS_ASSERT_THROWS_NOTHING(correction.initialize())
+    TS_ASSERT(correction.isInitialized())
+  }
 
-    Mantid::API::AnalysisDataService::Instance().remove(inputWS);
-    Mantid::API::AnalysisDataService::Instance().remove(outputWS);
+  void testExec() {
+    const std::string &inputWS = "inputWS";
+    setupWorkspaceForExecTest(inputWS);
+    runExecTest(inputWS, correction);
+  }
+
+  /*
+     * Function that will validate results against known results found with
+     * "standard" HFIR reduction package.
+     */
+  void testValidate() {
+    const std::string inputWS = "validateTest";
+    // sets up the workspace for the validate test
+    setupWorkspaceFileForValidateTest(inputWS);
+    runValidateTest(inputWS, correction);
   }
 
 private:
   Mantid::Algorithms::CalculateEfficiency correction;
-  std::string inputWS;
 };
 
+class CalculateEfficiencyTestPerformance : public CxxTest::TestSuite {
+public:
+  static CalculateEfficiencyTestPerformance *createSuite() {
+    return new CalculateEfficiencyTestPerformance();
+  }
+  static void destroySuite(CalculateEfficiencyTestPerformance *suite) {
+    delete suite;
+  }
+
+  CalculateEfficiencyTestPerformance() {
+
+    // This is in the constructor because it needs to
+    // run only ONCE, if put in setUp() it will be run before
+    // each test
+    execTestInputWs = "execTestInputWs";
+    setupWorkspaceForExecTest(execTestInputWs);
+    validateTestInputWs = "validateTestInputWs";
+    setupWorkspaceFileForValidateTest(validateTestInputWs);
+  }
+  void setUp() override {
+    if (!correction.isInitialized())
+      correction.initialize();
+  }
+
+  void tearDown() override {}
+
+  void testExecPerformance() {
+    // Workspaces are removed in the test
+    runExecTest(execTestInputWs, correction, performance);
+  }
+
+  void testValidatePerformance() {
+    // Workspaces are removed in the test
+    runValidateTest(validateTestInputWs, correction, performance);
```

Comment:
Yes, they are all disabled because of the `performance` boolean parameter that skips over all of them on the condition [here `  if (!performance) {...}`](https://github.com/mantidproject/mantid/pull/17506/files/cf4d917f190d16774d7e9f9c3efb172cdbd2e760#diff-5c63d294008d21cc146ffa517c97b43bR191)